fifteen_min/src/viewer.rs

```diff
@@ -66,14 +65,46 @@ impl State<SimpleApp> for Viewer {
 
         if ctx.redo_mouseover() {
             self.hovering_on_bldg = match app.mouseover_unzoomed_buildings(ctx) {
-                Some(ID::Building(id)) => Some(HoverOnBuilding {
-                    id,
-                    tooltip: if let Some(time) = self.isochrone.time_to_reach_building.get(&id) {
-                        Text::from(Line(format!("{} away", time)))
-                    } else {
-                        Text::from(Line("This is more than 15 minutes away"))
-                    },
-                }),
+                Some(ID::Building(hover_id)) => match self.hovering_on_bldg.take() {
+                    Some(previous_hover) if previous_hover.id == hover_id => Some(previous_hover),
+                    _ => {
+                        debug!("drawing new hover");
+                        let drawn_route = self
+                            .isochrone
+                            .path_to(&app.map, hover_id)
+                            .and_then(|path| path.trace(&app.map, Distance::ZERO, None))
+                            .map(|polyline| {
+                                let scale_factor =
+                                    if ctx.canvas.cam_zoom >= app.opts.min_zoom_for_detail {
+                                        1.0
+                                    } else {
+                                        10.0
+                                    };
+                                let dashed_lines = polyline.dashed_lines(
+                                    Distance::meters(0.75 * scale_factor),
+                                    Distance::meters(1.0 * scale_factor),
+                                    Distance::meters(0.4 * scale_factor),
+                                );
+                                let colored_lines: Vec<(Color, Polygon)> = dashed_lines
```

Comment:
Alternatively, `let mut batch = GeomBatch::new(); batch.extend(Color::BLACK, dashed_lines);`

Reply:
Done! And pushed up a fix to redraw the hover when changing zoom state without changing buildings.